File: BookingAccomodation/Accomodations/Model/AvailablePeriod.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace reservation_service.Model
{
    public enum TypeOfPrice
    {
        ByGuest,
        ByAccomodationUnit
    }

    public class AvailablePeriod
    {
        [BsonRepresentation(BsonType.String)] public Guid Id { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public int Price { get; set; }
        public TypeOfPrice Type { get; set; }
        [BsonRepresentation(BsonType.String)] public Guid AccomodationId { get; set; }
    }
}

File: frontend-booking/src/app/modules/booking/model/availablePeriod.ts
export class AvailablePeriod{
  start: Date = new Date();
  end: Date = new Date();
  price: number = 0;
  type: number = 0;
  accomodationId: string = '';


  constructor(obj?: any) {
    if (obj) {
      this.start = obj.start ? new Date(obj.start) : new Date();
      this.end = obj.end ? new Date(obj.end) : new Date();
      this.price = obj.price;
      this.type = obj.type;
      this.accomodationId = obj.accomodationId;
    }
  }
}


